public/app/features/alerting/unified/components/rules/RuleDetailsActionButtons.tsx

```diff
@@ -263,6 +267,25 @@ export const RuleDetailsActionButtons: FC<Props> = ({ rule, rulesSource, isViewM
   return null;
 };
 
+/**
+ * We don't want to show the silence button if either
+ * 1. the user has no permissions to create silences
+ * 2. the admin has configured to only send instances to external AMs
+ */
+function useCanSilence(alertmanagerSourceName?: string) {
```

Comment:
I think we should refactor this a bit. We show the `Silence` button only for Grafana-managed alert rules and the condition in https://github.com/grafana/grafana/pull/62133/files#diff-691394a56baf1c8cb3534abb9530db44e1329d6f0bd8866049404848779bd13cR47-R49 is no longer valid
I think we could get rid of `alertmanagerSourceName` and only check if this is a Grafana-managed rule

Reply:
Yea that makes sense, will update the hook to check the type of the rule instead 👍